src/main/kotlin/sh/hsp/techtree/FileSystemYamlReader.kt

```diff
@@ -5,15 +5,65 @@ import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
 import com.fasterxml.jackson.module.kotlin.readValue
 import com.fasterxml.jackson.module.kotlin.registerKotlinModule
 import java.io.File
+import java.io.IOException
 import java.net.URL
 
 class FileSystemYamlReader : YamlReader {
     override fun readToModel(resource: URL): TreeModel {
         val mapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
         val file = File(resource.toURI())
         val yamlData = file.readText()
+        val treeModel: TreeModel = mapper.readValue(yamlData)
 
-        return mapper.readValue(yamlData)
+        validateTreeModel(treeModel)
+
+        return treeModel
```

Comment:
Ogólnie kotlin wykorzystuje ostatnią wartość jako return.
to można napisać, też jako:
```
readToModel(resource: URL) = File(resource.toURI())
.let { mapper.readValue(it) }
.also { validateTreeModel(it) }
```

Czy jakoś tak